src/InertialSense.cpp

```diff
@@ -1039,10 +1039,8 @@ bool InertialSense::isFirmwareUpdateSuccessful() {
     for (int i = 0; i < (int) m_comManagerState.devices.size(); i++) {
         ISFirmwareUpdater *fwUpdater = m_comManagerState.devices[i].fwUpdater;
         if (fwUpdater != nullptr) {
-            if (fwUpdater->hasPendingCommands() || (fwUpdater->fwUpdate_getSessionStatus() < fwUpdate::NOT_STARTED))
+            if (fwUpdater->hasPendingCommands() || ((fwUpdater->fwUpdate_getSessionStatus() >= fwUpdate::NOT_STARTED) && (fwUpdater->fwUpdate_getSessionStatus() < fwUpdate::FINISHED)))
```

Comment:
This should also be updated to use `if (fwUpdater != null && !fwUpdate->fwUpdate_isDone()) return false;` and ensure we are using the same logic in all places.  Note this one is a NOT fwUpdate_isDone().